File: apps/mobile/lib/features/community/domain/models/transport_task.dart
import 'package:meta/meta.dart';

enum TransportTaskStatus {
  open,
  accepted,
  pickedUp,
  inTransit,
  delivered,
  completed,
  cancelled,
}

enum TransportOfferStatus { pending, accepted, rejected }

@immutable
class TransportTask {
  final String id;
  final String? transportPostId;
  final String parentPostId;
  final int requesterId;
  final int? transporterId;
  final String pickupAddress;
  final double pickupLat;
  final double pickupLon;
  final String dropoffAddress;
  final double dropoffLat;
  final double dropoffLon;
  final DateTime? pickupWindowStart;
  final DateTime? pickupWindowEnd;
  final bool isSelfCollection;
  final TransportTaskStatus status;
  final DateTime? completedAt;
  final String? notes;
  final DateTime createdAt;
  final DateTime updatedAt;

  const TransportTask({
    required this.id,
    this.transportPostId,
    required this.parentPostId,
    required this.requesterId,
    this.transporterId,
    required this.pickupAddress,
    required this.pickupLat,
    required this.pickupLon,
    required this.dropoffAddress,
    required this.dropoffLat,
    required this.dropoffLon,
    this.pickupWindowStart,
    this.pickupWindowEnd,
    required this.isSelfCollection,
    required this.status,
    this.completedAt,
    this.notes,
    required this.createdAt,
    required this.updatedAt,
  });
}

@immutable
class TransportOffer {
  final String id;
  final String transportTaskId;
  final int transporterId;
  final String? message;
  final DateTime proposedPickupStart;
  final DateTime proposedPickupEnd;
  final TransportOfferStatus status;
  final DateTime createdAt;

  const TransportOffer({
    required this.id,
    required this.transportTaskId,
    required this.transporterId,
    this.message,
    required this.proposedPickupStart,
    required this.proposedPickupEnd,
    required this.status,
    required this.createdAt,
  });
}

File: apps/api/HappyPaws.Domain/Entities/TransportTask.cs
using System;
using System.Collections.Generic;
using HappyPaws.Domain.Enums;

namespace HappyPaws.Domain.Entities;

public class TransportTask : IAuditableEntity
{
    public Guid Id { get; set; }
    public Guid? TransportPostId { get; set; }
    public Guid ParentPostId { get; set; }
    public long RequesterId { get; set; }
    public long? TransporterId { get; set; }
    public string PickupAddress { get; set; } = null!;
    public NetTopologySuite.Geometries.Point PickupPoint { get; set; } = null!;
    public string DropoffAddress { get; set; } = null!;
    public NetTopologySuite.Geometries.Point DropoffPoint { get; set; } = null!;
    public DateTimeOffset? PickupWindowStart { get; set; }
    public DateTimeOffset? PickupWindowEnd { get; set; }
    public bool IsSelfCollection { get; set; } = false;
    public TransportTaskStatus Status { get; set; } = TransportTaskStatus.Open;
    public DateTimeOffset? CompletedAt { get; set; }
    public string? Notes { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public Post? TransportPost { get; set; }
    public Post ParentPost { get; set; } = null!;
    public User Requester { get; set; } = null!;
    public User? Transporter { get; set; }

    private readonly List<TransportOffer> _offers = new();
    public IReadOnlyCollection<TransportOffer> Offers => _offers.AsReadOnly();
}
